src/stdlib/io.rs

```diff
@@ -69,4 +88,831 @@ pub fn load_io<'gc>(ctx: Context<'gc>) {
+                    .into());
+            }
+
+            let file: Value = stack.get(0);
+            let file = if let Value::UserData(file) = file {
+                if let Ok(file) = file.downcast_static::<IoFile>() {
+                    file
+                } else {
+                    return Err("bad argument #1 to 'lines' (file expected)"
+                        .into_value(ctx)
+                        .into());
+                }
+            } else {
+                return Err("bad argument #1 to 'lines' (file expected)"
+                    .into_value(ctx)
+                    .into());
+            };
+
+            let values = if stack.is_empty() {
+                vec![ctx.intern(b"l").into_value(ctx)] // default format
+            } else {
+                stack.into_iter().skip(1).collect::<Vec<_>>()
+            };
+
+            let formats = values
+                .into_iter()
+                .enumerate()
+                .map(|(n, value)| {
+                    let Some(format) = value.into_string(ctx) else {
+                        return Err(format!(
+                            "bad argument #{} to 'lines' (string expected, got {})",
+                            n + 1,
+                            value.type_name()
+                        )
+                        .into_value(ctx)
+                        .into());
+                    };
+                    match format.to_str() {
+                        Ok(s) => Ok(s.to_string()),
+                        Err(err) => Err(err.to_string().into_value(ctx).into()),
+                    }
+                })
+                .collect::<Result<Vec<_>, Error<'_>>>()?;
+
+            let root = Lines {
+                position: RefCell::new(0),
+                file: file.clone(),
+                formats,
+            };
+
+            let lines = Callback::from_fn_with(&ctx, root, |root, ctx, _, _| {
+                Ok(CallbackReturn::Sequence(BoxSequence::new(
+                    &ctx,
+                    root.clone(),
+                )))
+            });
+
+            stack.replace(ctx, lines);
+            Ok(CallbackReturn::Return)
+        }),
+    );
+
+    file.set_field(
+        ctx,
+        "seek",
+        Callback::from_fn(&ctx, |ctx, _, mut stack| {
+            if stack.is_empty() {
+                return Err("bad argument #1 to 'seek' (file expected)"
+                    .into_value(ctx)
+                    .into());
+            }
+
+            let (file, whence, offset) =
+                stack.consume::<(UserData, Option<String>, Option<i64>)>(ctx)?;
+            let file = if let Ok(file) = file.downcast_static::<IoFile>() {
+                file
+            } else {
+                return Err("bad argument #1 to 'seek' (file expected)"
+                    .into_value(ctx)
+                    .into());
+            };
+
+            let whence_str = whence.map(|s| s.to_str().unwrap_or("cur")).unwrap_or("cur");
```

Comment:
This could be written as `whence.and_then(|s| s.to_str()).unwrap_or("cur")`, but it would be better to leave it was a byte string (rather than treating invalid utf8 as `"cur"`) -- so `whence.map(|s| s.as_bytes()).unwrap_or(b"cur")`.